### PR TITLE
citra_qt: Add Game List configuration

### PR DESCRIPTION
**Pull request by zhaowenlan1779**
This adds a Game List configuration group box which is similar to yuzu's, with features including icon size setting, row 1/2 text, and ability to hide invalid titles (those without a valid SMDH). I also added a UI tab and moved the language and theme settings there.

Screenshot:
image
(Icon size set to Small, row 1 set to Title Name and row 2 set to Title ID)

image
(The new UI tab)

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4259)

### CODE REVIEW

**Comment by zhaowenlan1779** [comment]
Conflicts with something on canary

**Comment by wwylele** [comment]
Please resolve conflict

**Comment by jroweboy** [comment]
while you are adding this, mind doing this change as well? https://github.com/yuzu-emu/yuzu/pull/1383/files

**Comment by FearlessTobi** [comment]
@jroweboy Umm didn't we already port this?

**Comment by zhaowenlan1779** [comment]
@jroweboy that does not really apply to Citra. Both sizes of icons we are getting (24x24 and 48x48) are natively stored in the SMDH and we did not scale them at all.

File: src/citra_qt/ui_settings.h [hunk 1]
```diff
@@ -58,6 +58,12 @@ struct Values {
     // Discord RPC
     bool enable_discord_presence;
 
+    // Game List
+    int game_list_icon_size;
+    int game_list_row_1;
+    int game_list_row_2;
```
Comment (wwylele) [inline src/citra_qt/ui_settings.h]: Please use `enum` for these

Comment (zhaowenlan1779) [reply src/citra_qt/ui_settings.h]: @wwylele changed to enums

File: src/citra_qt/ui_settings.h [hunk 2]
```diff
@@ -31,6 +31,20 @@ struct GameDir {
     };
 };
 
+enum GameListIconSize {
```
Comment (wwylele) [inline src/citra_qt/ui_settings.h]: `enum class`

**Comment by SothoTalKer** [comment]
Looks like there is a small bug. Reproduceable with Canary 964:

Set Row 1 to File Name and Row 2 to Title ID.
After a restart the order is reversed.

**Comment by zhaowenlan1779** [comment]
@SothoTalKer mind retesting with Canary 965 which will be available soon?

**Comment by SothoTalKer** [comment]
@zhaowenlan1779 it is fixed now 👍

**Comment by zhaowenlan1779** [comment]
I will merge this in 24 hours if no more comments

**Pull request merged**